scripts/process.py

```diff
@@ -1,61 +1,70 @@
-    urllib.request.urlretrieve(source_url, downloaded)
+    urllib.request.urlretrieve(source_url, downloaded_csv_filepath)
+
+
+def rename(package: PackageWrapper):
+    package.pkg.descriptor['resources'][0]['name'] = 'gold-prices-monthly'
+    package.pkg.descriptor['resources'][0]['path'] = 'data/monthly.csv'
+    package.pkg.descriptor['resources'][1]['name'] = 'gold-prices-yearly'
+    package.pkg.descriptor['resources'][1]['path'] = 'data/yearly.csv'
+    yield package.pkg
+    res_iter = iter(package)
+    first: ResourceWrapper = next(res_iter)
+    second: ResourceWrapper = next(res_iter)
+    yield first.it
+    yield second.it
+    yield from package
+
 
-def extract():
+def fix_rows(rows):
+    # We skip top 5 rows and trim notes from the bottom. Also as dates are without day specified
+    # we are taking the first day in month by default
+    newrows = [[row[0] + '-01', row[1]] for row in rows[5:-1]]
+    return newrows
+
+
+def extract_gold_prices(frequency):
     reader = csv.reader(open(downloaded))
-    newrows = [ row for row in reader ]
-    # skip top 5 rows
-    newrows = newrows[5:]
-    # trim the notes from the bottom
-    newrows = newrows[:-1]
-    # fix up the data
-    # dates are 1968-06 without day ...
-    newrows = [ [row[0] + '-01', row[1]] for row in newrows ]
-
-    existing = []
-    if os.path.exists(outpath):
-        fo = open(outpath)
-        existing = [ row for row in csv.reader(fo) ]
-        fo.close()
-    
-    starter = newrows[0]
-    for idx,row in enumerate(existing):
-        if row[0] == starter[0]:
-            # remove all rows from here on
-            del existing[idx:]
-            break
-    # and now add in new data
-    outrows = existing + newrows
-    fo = open(outpath, 'w')
-    writer = csv.writer(fo)
-    writer.writerows(outrows)
-    fo.close()
+    rows = [row for row in reader]
+    fixed_rows = fix_rows(rows)
+    for row in fixed_rows:
+        if frequency == 'monthly':
+            yield {'Date': row[0], 'Price': row[1]}
+        elif frequency == 'yearly':
+            if row[0].split('-')[1] == '01':
+                yield {'Date': row[0], 'Price': row[1]}
 
-import os
-os.environ['http_proxy'] = ''
-def upload():
-    import datastore.client as c
-    dsurl = 'http://datahub.io/dataset/gold-prices/resource/b9aae52b-b082-4159-b46f-7bb9c158d013'
-    client = c.DataStoreClient(dsurl)
-    client.delete()
-    client.upload(outpath)
-
-
-if __name__ == '__main__':
-    logging.basicConfig(level=logging.DEBUG)
-    print('Downloading')
-    download()
-    print('Extracting and merging')
-    extract()
-    
+
+download_csv_resource()
+gold_price_flow = Flow(
+    add_metadata(
+        name="gold-prices",
+        title="Gold Prices",
+        homepage='http://www.bundesbank.de',
+        licenses="PDDL-1.0",
+        version="0.2.0"
+    ),
+    extract_gold_prices('monthly'),
+    extract_gold_prices('yearly'),
+    set_type('Date', type='date', format='any'),
```

Comment:
You need `resources=None` if you want to set types for all resources

Reply:
@Branko-Dj this also is not fixed

Reply:
FIXED! All resources' types set